backends-clickhouse/src/main/scala/io/glutenproject/execution/CHSortMergeJoinExecTransformer.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.execution
+
+import io.glutenproject.extension.ValidationResult
+import io.glutenproject.utils.CHJoinValidateUtil
+
+import org.apache.spark.sql.catalyst.expressions.{Expression, NamedExpression}
+import org.apache.spark.sql.catalyst.plans.JoinType
+import org.apache.spark.sql.execution.SparkPlan
+
+import io.substrait.proto.JoinRel
+
+case class CHSortMergeJoinExecTransformer(
+    leftKeys: Seq[Expression],
+    rightKeys: Seq[Expression],
+    joinType: JoinType,
+    condition: Option[Expression],
+    left: SparkPlan,
+    right: SparkPlan,
+    isSkewJoin: Boolean = false,
+    projectList: Seq[NamedExpression] = null)
+  extends SortMergeJoinExecTransformerBase(
+    leftKeys,
+    rightKeys,
+    joinType,
+    condition,
+    left,
+    right,
+    isSkewJoin,
+    projectList) {
+
+  override protected def doValidateInternal(): ValidationResult = {
+    val shouldFallback =
+      CHJoinValidateUtil.shouldFallback(joinType, left.outputSet, right.outputSet, condition)
+    if (shouldFallback) {
+      return ValidationResult.notOk("ch join validate fail")
+    }
+    if (
+      substraitJoinType == JoinRel.JoinType.JOIN_TYPE_LEFT_SEMI
+      || substraitJoinType == JoinRel.JoinType.JOIN_TYPE_ANTI
+    ) {
+      return ValidationResult
+        .notOk(s"Found unsupported join type of $joinType for CH backend SMJ")
+    }
+
```

Comment:
I think this also could be move into `CHJoinValidateUtil.shouldFallback`.  `substraitJoinType` comes from `joinType`